### ISSUE TITLE
Additional resource types

### ISSUE THREAD

**Issue by dnephin** [dnephin/dobi#1]
Current resources are: Image, Run, Volume, Alias

Some more possible resources are:

**Push**

Tag an image with one or more tags and push them to a registry
```yaml
push=stage:
  image: <image resource name>
  tags: [...]
```

**Pull Image**

Pull an image from a registry, which can be used in place of an Image resource.
```yaml
pull=golang:
  image: <image with a tag>
  when: <always | missing | time.duration>
```

**Compose**

Start an environment defined by a Compose file

```yaml
compose=acceptance:
  file: [docker-compose.yml]
  project: "acceptance-test-{USER}"
```

**Service**

Start a service which will continue to run for the entire duration of the execution, and will be shutdown before dobi exits.

```yaml
service=mysql:
  use: <image resource name>
  ... # service options
```

~~**Task Alias**~~   Implemented in ecae1cd642e787df1958757da26fb13fad12c970

**Comment by dnephin** [comment]
Push and Pull were folded into image with the multi-task changes.

Compose is in v0.1.

Service has to wait on docker 1.12 release
